doc/developer/design/20230216_role_based_access_control.md

```diff
@@ -0,0 +1,368 @@
+
+### Phase 3 - `PUBLIC` role
+
+See [Grant](https://www.postgresql.org/docs/current/sql-grant.html) for PostgreSQL `PUBLIC` details (grep for PUBLIC).
+
+`PUBLIC` is a special keyword that is accepted anywhere a role name would be accepted. The key word PUBLIC indicates
+that the changes are to be applied to all roles, including those that might be created later.
+
+### Phase 4 - Privileges
+
+See [Privileges](https://www.postgresql.org/docs/current/ddl-priv.html) for PostgreSQL privileges.
+
+Roles can be granted and revoked certain privileges on objects, that allow them to perform some action with that object.
+
+We will support the following privileges:
+
+| Privilege | Description                                                              | Abbreviation | Applicable Object Types                       | From PostgreSQL |
+|-----------|--------------------------------------------------------------------------|--------------|-----------------------------------------------|-----------------|
+| `SELECT`  | Allows reading rows from an object.                                      | r(”read”)    | Table, View, Materialized View, Source        | Yes             |
+| `INSERT`  | Allows inserting into an object.                                         | a(”append”)  | Table                                         | Yes             |
+| `UPDATE`  | Allows updating an object (requires SELECT if a read is necessary).      | w(”write”)   | Table                                         | Yes             |
+| `DELETE`  | Allows deleting from an object (requires SELECT if a read is necessary). | d            | Table                                         | Yes             |
+| `CREATE`  | Allows creating a new object within another object.                      | C            | Database, Schema, Cluster                     | Yes             |
+| `USAGE`   | Allows using an object or looking up members of an object.               | U            | Database, Schema, Connection, Secret, Cluster | Yes             |
+
+We will support the following object types:
+
+| Object Type          | All Privileges | From PostgreSQL |
+|----------------------|----------------|-----------------|
+| `DATABASE`           | UC             | Yes             |
+| `SCHEMA`             | UC             | Yes             |
+| `TABLE`              | arwd           | Yes             |
+| `VIEW`               | r              | Yes             |
+| `MATERIALIZED  VIEW` | r              | Yes             |
+| `INDEX`              |                | Yes             |
+| `SOURCE`             | r              | No              |
+| `SINK`               |                | No              |
+| `CONNECTION`         | U              | No              |
+| `SECRET`             | U              | No              |
+| `CLUSTER`            | UC             | No              |
+
+When any of these objects are created, the creating role is assigned as the owner of that object. Only the owner (and
```

Comment:
Does this imply that only the super user will be able to manage the default cluster?

Reply:
That's a good question. There can be one or more super users, so any role with the super user attribute will be able to manage the default cluster, including re-assigning it's owner. I think it makes sense to assign `mz_system` as the initial owner of the default cluster.

Reply:
I added a couple of comments about this in the most recent commit: https://github.com/MaterializeInc/materialize/pull/17702/commits/c58d42e7666482d21b8920c446177374ad012354

Reply:
This makes sense to me too.